app/controllers/coffee_shop_controller.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require 'sinatra/base'
+require 'json'
+require_relative '../services/coffee_shop_finder_service'
+
+class CoffeeShopController < Sinatra::Base
+  configure do
+    set :protection, except: [:host_authorization]
+    set :show_exceptions, false
+    set :raise_errors, true
+    enable :logging
+  end
+
+  before do
+    content_type :json
+  end
+
+  get '/closest_shops' do
+    content_type 'text/plain'
+    lat, lon = validate_coordinates!(params)
+
+    shops_with_distances = coffee_shop_finder_service.closest_shops(lat, lon)
+    format_response(shops_with_distances, lat, lon)
+  rescue StandardError => e
+    handle_error(e)
+  end
+
+  private
+
+  # Format shops into "Name --> distance <-- (user-lat, user_lon)" strings
```

Comment:
Why this format? What's the intended audience?

Reply:
The format from that comment is outdated as well, it actually returns:
  ```
  Coffee shops nearest (user-lat, user_lon) by distance:

  distance <--> Name
  ```
I meant to match the `__Expected output__` in a more readable format